### PR TITLE
Add option to silence IF deprecation warning

### PR DESCRIPTION
**Pull request by larsoner**
Over in https://github.com/h5py/h5py/pull/2444 the GitHub actions logs get clogged with hundreds/thousands of:
```
  warning: h5py/api_types_hdf5.pxd:710:2: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
```
The proper solution is to fix these. But in the meantime, it would be nice if I could:

1. Open a GitHub issue about it for `h5py` to keep track of it with milestones etc., and
2. Silence the warnings in the meantime so the build logs are more readable

This PR moves toward (2). Consider it a proposal-as-PR so feel free to close if it's not the way to go -- I wanted to see if it was doable *at all* and it turned out not to be too bad to get something that worked for h5py at least, where we could with this PR do `cythonize(..., compiler_directives={'warn.deprecated_if': False})`.

### CODE REVIEW

**Comment by larsoner** [comment]
... and if this *does* seem worthwhile, I'm happy to look into the Cython contributing guide on how to clean this up / what else needs to be added.

File: Cython/Compiler/Parsing.py [hunk 1]
```diff
@@ -2528,6 +2528,7 @@ def p_IF_statement(s: PyrexScanner, ctx):
 @cython.cfunc
 def p_statement(s: PyrexScanner, ctx, first_statement: cython.bint = 0):
     cdef_flag = ctx.cdef_flag
+    warn_deprecated_if = s.context.compiler_directives.get("warn.deprecated_if", False)
```
Comment (scoder) [inline Cython/Compiler/Parsing.py:2531]: Most people won't have `IF` in their code, so this should rather go into the `elif "IF"` block below.

File: docs/src/userguide/source_files_and_compilation.rst [hunk 2]
```diff
@@ -787,7 +787,7 @@ Cython code.  Here is the list of currently supported directives:
     Default is True.
 
     .. versionchanged:: 3.0.0
-        Default changed from False to True 
+        Default changed from False to True
```
Comment (scoder) [inline docs/src/userguide/source_files_and_compilation.rst:790]: This whole space cleanup should go into a separate commit outside of this PR.

File: Cython/Compiler/Options.py [hunk 3]
```diff
@@ -235,6 +235,7 @@ def copy_inherited_directives(outer_directives, **new_directives):
     'warn.unused_arg': False,
     'warn.unused_result': False,
     'warn.multiple_declarators': True,
+    'warn.deprecated_if': True,
```
Comment (scoder) [inline Cython/Compiler/Options.py:238]: Assuming that there might be more deprecation warnings to hide at some point, I propose this naming scheme:
```suggestion
    'warn.deprecated.IF': True,
```
And then we should probably also add `warn.deprecated.DEF`.

**Review by scoder — COMMENTED** [review]
Seems ok to add this.

File: Cython/Compiler/Options.py (hunk 3 above)
Comment (larsoner) [reply Cython/Compiler/Options.py:238]: Looks like the `DEF` is currently commented out, so probably no need to add it here?

Comment (scoder) [reply Cython/Compiler/Options.py]: I'd go the other way: if we can add an option to enable/disable that warning, then we should comment it back in and disable the warning using the new option instead.

Comment (larsoner) [reply Cython/Compiler/Options.py]: Done!

**Comment by da-woods** [comment]
Thanks
